Serialization failures with boost-1.65.0

With (almost) newest boost 1.65.0 I get a number of serialization failures [1]

```
In file included from /usr/include/boost/serialization/vector.hpp:26:0,
                 from /srv/testsuite/dealii/include/deal.II/base/index_set.h:23,
                 from /srv/testsuite/dealii/include/deal.II/lac/vector.h:24,
                 from /srv/testsuite/dealii/include/deal.II/lac/petsc_vector_base.h:26,
                 from /srv/testsuite/dealii/tests/distributed_grids/../tests.h:74,
                 from /srv/testsuite/dealii/tests/distributed_grids/grid_tools_exchange_cell_data_01.cc:19:
/usr/include/boost/serialization/access.hpp: In instantiation of ‘static void boost::serialization::access::serialize(Archive&, T&, unsigned int) [with Archive = boost::archive::binary_oarchive; T = std::array<unsigned int, 4ul>]’:
/usr/include/boost/serialization/serialization.hpp:68:69:   required from ‘void boost::serialization::serialize(Archive&, T&, unsigned int) [with Archive = boost::archive::binary_oarchive; T = std::array<unsigned int, 4ul>]’
/usr/include/boost/serialization/serialization.hpp:126:23:   required from ‘void boost::serialization::serialize_adl(Archive&, T&, unsigned int) [with Archive = boost::archive::binary_oarchive; T = std::array<unsigned int, 4ul>]’
/usr/include/boost/archive/detail/oserializer.hpp:154:5:   required from ‘void boost::archive::detail::oserializer<Archive, T>::save_object_data(boost::archive::detail::basic_oarchive&, const void*) const [with Archive = boost::archive::binary_oarchive; T = std::array<unsigned int, 4ul>]’
/usr/include/boost/archive/detail/oserializer.hpp:103:1:   required from ‘class boost::archive::detail::oserializer<boost::archive::binary_oarchive, std::array<unsigned int, 4ul> >’
/usr/include/boost/archive/detail/oserializer.hpp:255:13:   required from ‘static void boost::archive::detail::save_non_pointer_type<Archive>::save_standard::invoke(Archive&, const T&) [with T = std::array<unsigned int, 4ul>; Archive = boost::archive::binary_oarchive]’
/usr/include/boost/archive/detail/oserializer.hpp:310:28:   [ skipping 22 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/archive/binary_oarchive_impl.hpp:59:9:   required from ‘void boost::archive::binary_oarchive_impl<Archive, Elem, Tr>::save_override(T&) [with T = const dealii::parallel::GridTools::internal::CellDataTransferBuffer<2, double>; Archive = boost::archive::binary_oarchive; Elem = char; Tr = std::char_traits<char>]’
/usr/include/boost/archive/detail/interface_oarchive.hpp:70:9:   required from ‘Archive& boost::archive::detail::interface_oarchive<Archive>::operator<<(const T&) [with T = dealii::parallel::GridTools::internal::CellDataTransferBuffer<2, double>; Archive = boost::archive::binary_oarchive]’
/srv/testsuite/dealii/include/deal.II/distributed/grid_tools.h:221:21:   required from ‘std::vector<char> dealii::parallel::GridTools::internal::CellDataTransferBuffer<dim, T>::pack_data() const [with int dim = 2; T = double]’
/srv/testsuite/dealii/include/deal.II/distributed/grid_tools.h:353:28:   required from ‘void dealii::parallel::GridTools::exchange_cell_data_to_ghosts(const MeshType&, std::function<boost::optional<T>(const typename MeshType::active_cell_iterator&)>, std::function<void(const typename MeshType::active_cell_iterator&, const DataType&)>) [with DataType = double; MeshType = dealii::parallel::distributed::Triangulation<2, 2>; typename MeshType::active_cell_iterator = dealii::TriaActiveIterator<dealii::CellAccessor<2, 2> >]’
/srv/testsuite/dealii/tests/distributed_grids/grid_tools_exchange_cell_data_01.cc:70:4:   required from ‘void test() [with int dim = 2]’
/srv/testsuite/dealii/tests/distributed_grids/grid_tools_exchange_cell_data_01.cc:83:12:   required from here
/usr/include/boost/serialization/access.hpp:116:9: error: ‘struct std::array<unsigned int, 4ul>’ has no member named ‘serialize’
         t.serialize(ar, file_version);
         ^
/usr/include/boost/serialization/access.hpp: In instantiation of ‘static void boost::serialization::access::serialize(Archive&, T&, unsigned int) [with Archive = boost::archive::binary_iarchive; T = std::array<unsigned int, 4ul>]’:
/usr/include/boost/serialization/serialization.hpp:68:69:   required from ‘void boost::serialization::serialize(Archive&, T&, unsigned int) [with Archive = boost::archive::binary_iarchive; T = std::array<unsigned int, 4ul>]’
/usr/include/boost/serialization/serialization.hpp:126:23:   required from ‘void boost::serialization::serialize_adl(Archive&, T&, unsigned int) [with Archive = boost::archive::binary_iarchive; T = std::array<unsigned int, 4ul>]’
/usr/include/boost/archive/detail/iserializer.hpp:192:5:   required from ‘void boost::archive::detail::iserializer<Archive, T>::load_object_data(boost::archive::detail::basic_iarchive&, void*, unsigned int) const [with Archive = boost::archive::binary_iarchive; T = std::array<unsigned int, 4ul>]’
/usr/include/boost/archive/detail/iserializer.hpp:120:1:   required from ‘class boost::archive::detail::iserializer<boost::archive::binary_iarchive, std::array<unsigned int, 4ul> >’
/usr/include/boost/archive/detail/iserializer.hpp:410:13:   required from ‘static void boost::archive::detail::load_non_pointer_type<Archive>::load_standard::invoke(Archive&, const T&) [with T = std::array<unsigned int, 4ul>; Archive = boost::archive::binary_iarchive]’
/usr/include/boost/archive/detail/iserializer.hpp:462:28:   [ skipping 22 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/archive/binary_iarchive_impl.hpp:58:9:   required from ‘void boost::archive::binary_iarchive_impl<Archive, Elem, Tr>::load_override(T&) [with T = dealii::parallel::GridTools::internal::CellDataTransferBuffer<2, double>; Archive = boost::archive::binary_iarchive; Elem = char; Tr = std::char_traits<char>]’
/usr/include/boost/archive/detail/interface_iarchive.hpp:68:9:   required from ‘Archive& boost::archive::detail::interface_iarchive<Archive>::operator>>(T&) [with T = dealii::parallel::GridTools::internal::CellDataTransferBuffer<2, double>; Archive = boost::archive::binary_iarchive]’
/srv/testsuite/dealii/include/deal.II/distributed/grid_tools.h:251:19:   required from ‘void dealii::parallel::GridTools::internal::CellDataTransferBuffer<dim, T>::unpack_data(const std::vector<char>&) [with int dim = 2; T = double]’
/srv/testsuite/dealii/include/deal.II/distributed/grid_tools.h:379:11:   required from ‘void dealii::parallel::GridTools::exchange_cell_data_to_ghosts(const MeshType&, std::function<boost::optional<T>(const typename MeshType::active_cell_iterator&)>, std::function<void(const typename MeshType::active_cell_iterator&, const DataType&)>) [with DataType = double; MeshType = dealii::parallel::distributed::Triangulation<2, 2>; typename MeshType::active_cell_iterator = dealii::TriaActiveIterator<dealii::CellAccessor<2, 2> >]’
/srv/testsuite/dealii/tests/distributed_grids/grid_tools_exchange_cell_data_01.cc:70:4:   required from ‘void test() [with int dim = 2]’
/srv/testsuite/dealii/tests/distributed_grids/grid_tools_exchange_cell_data_01.cc:83:12:   required from here
/usr/include/boost/serialization/access.hpp:116:9: error: ‘struct std::array<unsigned int, 4ul>’ has no member named ‘serialize’
```

@tjhei *ping*

[1] https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=11385

--- a/include/deal.II/distributed/grid_tools.h
+++ b/include/deal.II/distributed/grid_tools.h
@@ -25,6 +25,7 @@ DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/optional.hpp>
 #include <boost/archive/binary_oarchive.hpp>
 #include <boost/archive/binary_iarchive.hpp>
+#include <boost/serialization/vector.hpp>
 #include <boost/iostreams/stream.hpp>
 #include <boost/iostreams/filtering_stream.hpp>
 #include <boost/iostreams/device/back_inserter.hpp>

